src/Http/Routing/test/UnitTests/Constraints/FIleNameRouteConstraintTest.cs

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Xunit;
+
+namespace Microsoft.AspNetCore.Routing.Constraints
+{
+    public class FileNameRouteConstraintTest
+    {
+        public static TheoryData<object> FileNameData
+        {
+            get
+            {
+                return new TheoryData<object>()
+                {
+                    "hello.txt",
+                    "hello.txt.jpg",
+                    "/hello.t",
+                    "/////hello.x",
+                    "a/b/c/d.e",
+                    "a/b./.c/d.e",
+                };
```

Comment:
I would also add a leading dot combinations like `.gitignore`

Reply:
On balance I agree. Leading-dotted things probably are filenames, and are not likely to be desired as part of the client-side URL space. It's a weird scenario, though.

Reply:
yeah on reflection, I agree. I'll update this.